### PR TITLE
🌱 Bump controller-runtime to v0.18.0

### PR DESCRIPTION
**Pull request by chrischdi**
**What this PR does / why we need it**:

WIP PR, bumps CR to current main for testing purposes

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by sbueringer** [comment]
Thx! lgtm so far

**Comment by chrischdi** [comment]
Test run for:
- #10026

**Comment by chrischdi** [comment]
/retest

x509 flake

File: controllers/external/tracker.go [hunk 1]
```diff
@@ -54,15 +53,13 @@ func (o *ObjectTracker) Watch(log logr.Logger, obj runtime.Object, handler handl
...
 
-	u := &unstructured.Unstructured{}
-	u.SetGroupVersionKind(gvk)
-
 	log.Info(fmt.Sprintf("Adding watch on external object %q", gvk.String()))
-	err := o.Controller.Watch(
-		source.Kind(o.Cache, u),
+	err := o.Controller.Watch(source.Kind(
+		o.Cache,
+		obj,
```
Comment (sbueringer) [inline controllers/external/tracker.go:59]: let's deepcopy this object to be safe

File: go.mod [hunk 2]
```diff
@@ -35,18 +35,18 @@ require (
...
+	k8s.io/apiserver v0.30.0
+	k8s.io/client-go v0.30.0
 	k8s.io/cluster-bootstrap v0.29.3
-	k8s.io/component-base v0.29.3
-	k8s.io/klog/v2 v2.110.1
-	k8s.io/kube-openapi v0.0.0-20231010175941-2dd684a91f00
+	k8s.io/component-base v0.30.0
+	k8s.io/klog/v2 v2.120.1
+	k8s.io/kube-openapi v0.0.0-20240228011516-70dd3763d340
 	k8s.io/kubectl v0.29.3
```
Comment (sbueringer) [inline go.mod:47]: Please also grep for other v0.29.x dependencies and bump them

**Comment by sbueringer** [comment]
@Danil-Grigorev This PR contains the minimal amount of changes to bump to the new CR version. Do you want to rebase your PR (https://github.com/kubernetes-sigs/cluster-api/pull/10464) after this PR merges to leverage the new CR generic types more broadly?

**Comment by sbueringer** [comment]
Thx!

/lgtm
/approve

**Comment by sbueringer** [comment]
/hold cancel

Thought I missed something, but all good

**Pull request merged**